riff-raff/app/persistence/mongodb.scala

```diff
@@ -184,9 +196,11 @@ class MongoDatastore(database: MongoDB) extends DataStore with DocumentStore wit
       deployCollection.findOneByID(uuid).flatMap(DeployRecordDocument.fromDBO(_))
     }
 
-  override def writeLog(log: LogDocument) {
-    logAndSquashExceptions(Some("Writing new log document with id %s for deploy %s" format (log.id, log.deploy)),()) {
-      deployLogCollection.insert(log.toDBO, WriteConcern.Safe)
+  override def writeLog(log: LogDocument) = {
+    retry(MAX_RETRIES) { 
+      logExceptions(Some("Writing new log document with id %s for deploy %s" format (log.id, log.deploy))) {
```

Comment:
I suspect that the way this composition currently does logging might not give us as much visibility as we might want. The logging will not clearly inform us if a retry eventually succeeded or not - or how many retries were attempted. I'd consider baking that into the `retry` function.

Reply:
Yes, good idea 👍